### PR TITLE
Email widget support

### PR DESCRIPTION
**Pull request by chenliangde**
Note: It works fine in all tested email clients:  
> ```<div style="display:none">...```

 
Usage:

```
<div {{block class="Nosto\Tagging\Block\Email\Visible"}} >
  <img src="{{block class="Nosto\Tagging\Block\Email\ImageUrl" url_template="http://localhost:9000/image/v1/@NOSTO_ACCOUNT@/@CAMPAIGN_ID@/1?uid=@EMAIL@&amp;version=2.0.8" customer_email=$order-getCustomerEmail() campainId="bestseller" }}"></img>
</div>
```

### CODE REVIEW

File: Block/ImageUrl.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    public function _toHtml()
+    {
+        $store = $this->nostoHelperScope->getStore(true);
+        $account = $this->nostoHelperAccount->findAccount($store);
+
+        if (!$account || !$account->getName()) {
+            return '';
+        }
+
+        $urlTemplate = $this->getData(self::URL_TEMPLATE);
```
Comment (hannunosto) [inline Block/ImageUrl.php]: There's quite some copy-paste between this block and https://github.com/Nosto/nosto-magento2/pull/318/files#diff-017894b02e75056a2c9d418c14f6b9d1R86. Can we use trait or somehow get around copy paste?

**Comment by hannunosto** [comment]
Things to modify
* `email_template` attribute should be optional and the block should build a default one by using `NOSTO_SERVER_URL` env variable
* add `recommendation_type` as an attribute - this will be the reco / algo type

File: Jenkinsfile [hunk 2]
```diff
@@ -43,7 +43,6 @@ pipeline {
         sh "composer create-project magento/community-edition magento"
         sh "cd magento && composer config minimum-stability dev"
         sh "cd magento && composer config prefer-stable true"
-        sh "cd magento && composer config repositories.2 git https://github.com/${env.CHANGE_FORK}/nosto-magento2"
```
Comment (supercid) [inline Jenkinsfile]: Can you revert this line since the issue was a temporary glitch in Jenkins?

Comment (chenliangde) [reply Jenkinsfile]: Reverted

**Pull request merged**